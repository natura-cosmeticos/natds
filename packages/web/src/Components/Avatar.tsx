import React, { FunctionComponent } from 'react';
import MaterialAvatar, { AvatarProps as MaterialAvatarProps } from '@material-ui/core/Avatar';

import { withTheme } from '@material-ui/styles';
import { IAvatarSizes } from '@naturacosmeticos/natds-styles';
import { IThemeWeb } from 'Themes';

export type AvatarSizes = 'tiny' | 'small' | 'standard' | 'large' | 'xlarge';
export type AvatarColors = 'primary' | 'secondary' | string;

interface IAvatarProps extends MaterialAvatarProps {
  size?: AvatarSizes;
  color?: AvatarColors;
  theme: IThemeWeb | unknown;
}

const Avatar: FunctionComponent<IAvatarProps> = (props: IAvatarProps) => {
  const style = {
    ...withSizes(props.size, props.theme),
    ...withColors(props.color, props.theme)
  };

  return <MaterialAvatar {...props} style={style} />;
};

function withSizes(size: AvatarSizes | undefined = 'standard', theme: any = {}) {
  const { avatarSizes }: { avatarSizes: IAvatarSizes } = theme;

  if (!avatarSizes) return {};

  const value = avatarSizes[size];
  const sizeInPx = `${value.size}px`;

  return {
    width: sizeInPx,
    height: sizeInPx,
    fontSize: value.fontSize
  };
}

function withColors(color: AvatarColors | undefined = 'primary', theme: any = {}) {
  const { palette = {} }: { palette: object } = theme;
  const selectedColor = palette[color];

  if (!selectedColor) return {};

  return {
    background: selectedColor.main
  };
}

export default withTheme(Avatar);
