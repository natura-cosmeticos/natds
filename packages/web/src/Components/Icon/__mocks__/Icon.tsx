import * as React from 'react'
import { IMockedIconProps } from './Icon.props'

export const Icon : React.FunctionComponent<IMockedIconProps> = (props: IMockedIconProps) => <i {...props} />

export default Icon
