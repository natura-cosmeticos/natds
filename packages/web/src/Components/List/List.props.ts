import * as React from 'react'
import { ListProps } from '@material-ui/core/List'

export interface IListProps extends ListProps {

  /**
   * The content of the component.
   *
   * @optional
   * @type node
   */
  children?: ListProps['children']

  /**
   * Override or extend the styles applied to the component.
   *
   * Check [CSS API](https://material-ui.com/api/list/#css) for more details.
   *
   * @optional
   * @see https://material-ui.com/api/list/#css
   * @type object
   */
  classes?: ListProps['classes']

  /**
   * @default "ul"
   * @optional
   * @type element
   */
  component?: React.ReactElement

  /**
   * If `true`, compact vertical padding designed for keyboard and mouse input will be used for the list and list items.
   *
   * The prop is available to descendant components as the dense context.
   *
   * @default false
   * @optional
   * @type bool
   */
  dense?: ListProps['dense']

  /**
   * If `true`, vertical padding will be removed from the list.
   *
   * @default false
   * @optional
   * @type bool
   */
  disablePadding?: ListProps['disablePadding']

  /**
   * The content of the subheader, normally `ListSubheader`.
   *
   * @optional
   * @see ListSubheader
   * @type node
   */
  subheader?: ListProps['subheader']
}
