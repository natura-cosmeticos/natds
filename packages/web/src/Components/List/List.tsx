import * as React from 'react'
import MaterialList, { ListProps as IListProps } from '@material-ui/core/List'

/**
 * For more advanced usages, check [Material UI List docs](https://material-ui.com/components/lists/)
 *
 * ## Importing
 *
 * ```
 * import { List } from '@naturacosmeticos/natds-web';
 * ```
 *
 * @see https://material-ui.com/components/lists/
 */
export const List = React.forwardRef<HTMLUListElement, IListProps>((
  props: IListProps,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref: any
) => <MaterialList {...props} ref={ref} />)

List.displayName = 'List'

export default List
