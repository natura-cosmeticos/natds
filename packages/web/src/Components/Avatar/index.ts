import Avatar from './Avatar'

export { IAvatarProps } from './Avatar.props'
export { Avatar }
export default Avatar
