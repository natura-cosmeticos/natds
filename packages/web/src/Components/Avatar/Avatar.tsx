import * as React from 'react'
import MaterialAvatar from '@material-ui/core/Avatar'
import makeStyles from '@material-ui/core/styles/makeStyles'
import { IAvatarProps } from './Avatar.props'
import { IThemeWeb } from '../../Themes'
import styles from './Avatar.styles'
import { Icon } from '../Icon'

/**
 * ## Importing
 *
 * For more advanced usages, check [Material UI Avatar docs](https://material-ui.com/components/avatars/)
 *
 * ```
 * import { Avatar } from "@naturacosmeticos/natds-web";
 * ```
 */
export const Avatar = React.forwardRef<HTMLDivElement, IAvatarProps>((
  props: IAvatarProps, ref
) => {
  const {
    color,
    size,

    /**
     * @deprecated `variant` property is deprecated since 0.19.0.
     * It will be removed on 1.0 and all Avatar variants will be `circle` by default.
     */
    variant,
    ...otherProps
  } = props

  const useStyles = makeStyles(
    (theme: IThemeWeb) => styles({ color, size, theme }), { name: 'NatDsAvatar' }
  )

  const classes = useStyles()

  return (
    <MaterialAvatar
      classes={classes}
      variant={variant}
      {...otherProps}
      ref={ref}
    >
      {props.children || <Icon name="outlined-social-person" />}
    </MaterialAvatar>
  )
})

Avatar.displayName = 'Avatar'
export default Avatar
