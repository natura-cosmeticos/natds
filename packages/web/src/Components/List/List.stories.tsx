/* eslint-disable no-magic-numbers */
import * as React from 'react'
import { Meta, Story } from '@storybook/react'
import { IListProps } from './List.props'
import { List } from './List'
import { ListSubheader } from '../ListSubheader'
import { ListItemText } from '../ListItemText'
import { ListItem } from '../ListItem'
import { ListItemIcon } from '../ListItemIcon'
import { ListItemAvatar } from '../ListItemAvatar'
import { ListItemSecondaryAction } from '../ListItemSecondaryAction'
import {
  WithAvatarAndTexts, WithBothPrimaryAndSecondaryTexts,
  WithCheckboxTextAndIconButton, WithIconAndText, WithIconTextAndCheckbox,
  WithIconTextAndSwitch, WithPrimaryTextOnly,
  WithTextAndSecondaryAction
} from '../ListItem/ListItem.stories'

export default {
  component: List,
  subcomponents: {
    ListItem,
    ListItemAvatar,
    ListItemIcon,
    ListItemSecondaryAction,
    ListItemText,
    ListSubheader
  },
  title: 'Components/List'
} as Meta

type Event = React.MouseEvent<HTMLElement>;

const Template : Story<IListProps> = (args: IListProps) => {
  const initialSelectedIndex = 0
  const [selectedIndex, setSelectedIndex] = React.useState<number>(initialSelectedIndex)

  const handleClick = (event: Event, index: number) => {
    setSelectedIndex(index)
  }

  const listItems = [
    WithPrimaryTextOnly,
    WithBothPrimaryAndSecondaryTexts,
    WithIconAndText,
    WithTextAndSecondaryAction,
    WithIconTextAndSwitch,
    WithCheckboxTextAndIconButton,
    WithIconTextAndCheckbox,
    WithAvatarAndTexts
  ]

  return (
    <List {...args}>
      {args.children || listItems.map((item: Story, index: number) => (
        <ListItem
          {...item.args}
          key={index}
          onClick={(event: Event) => handleClick(event, index)}
          selected={selectedIndex === index}
        />
      ))}
    </List>
  )
}

export const Playground : Story<IListProps> = Template.bind({})
Playground.args = {
  className: '',
  dense: false,
  disablePadding: false
}

export const ForNesting : Story<IListProps> = Template.bind({})
ForNesting.args = {
  ...Playground.args,
  children: <ListItem {...WithIconAndText.args} />,
  disablePadding: true
}
