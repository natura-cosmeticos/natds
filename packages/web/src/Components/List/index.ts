import List from './List'

export { IListProps } from './List.props'
export { List }
export default List
