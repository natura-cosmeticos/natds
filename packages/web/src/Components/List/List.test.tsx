import * as React from 'react'
import * as TestRenderer from 'react-test-renderer'
import { List } from './List'
import { ListSubheader } from '../ListSubheader'
import { ListItem } from '../ListItem'
import { ListItemAvatar } from '../ListItemAvatar'
import { Avatar } from '../Avatar'
import { Icon } from '../Icon'
import { ListItemIcon } from '../ListItemIcon'
import { ListItemSecondaryAction } from '../ListItemSecondaryAction'
import { IconButton } from '../IconButton'
import { ListItemText } from '../ListItemText'

describe('List components', () => {
  describe('rendering variants', () => {
    test('should match to snapshot - empty list', () => {
      const component = TestRenderer.create(<List />)

      expect(component).toMatchSnapshot('List empty snapshot')
    })

    test('should match to snapshot - with ListSubheader', () => {
      const component = TestRenderer.create(<List
        subheader={<ListSubheader id="subheader">Subheader</ListSubheader>}
      />)

      expect(component).toMatchSnapshot('List with ListSubheader snapshot')
    })

    test('should match to snapshot - with ListItem', () => {
      const component = TestRenderer.create(<List>
        <ListItem>Test</ListItem>
      </List>)

      expect(component).toMatchSnapshot('List with ListItem snapshot')
    })

    test('should match to snapshot - with ListItem button', () => {
      const component = TestRenderer.create(<List>
        <ListItem button>Test</ListItem>
      </List>)

      expect(component).toMatchSnapshot('List with ListItem button snapshot')
    })

    test('should match to snapshot - with ListItemAvatar', () => {
      const component = TestRenderer.create(<List>
        <ListItem>
          <ListItemAvatar>
            <Avatar color="default" size="small">
              <Icon name="outlined-navigation-close" size="tiny" />
            </Avatar>
          </ListItemAvatar>
        </ListItem>
      </List>)

      expect(component).toMatchSnapshot('List with ListItemAvatar snapshot')
    })

    test('should match to snapshot - with ListItemIcon', () => {
      const component = TestRenderer.create(<List>
        <ListItem>
          <ListItemIcon>
            <Icon name="outlined-navigation-close" size="small" />
          </ListItemIcon>
        </ListItem>
      </List>)

      expect(component).toMatchSnapshot('List with ListItemIcon snapshot')
    })

    test('should match to snapshot - with ListItemSecondaryAction', () => {
      const component = TestRenderer.create(<List>
        <ListItem>
          <ListItemSecondaryAction>
            <IconButton edge="end" aria-label="shopping">
              <Icon name="outlined-navigation-close" size="tiny" />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
      </List>)

      expect(component).toMatchSnapshot('List with ListItemSecondaryAction snapshot')
    })

    test('should match to snapshot - with ListItemText', () => {
      const component = TestRenderer.create(<List>
        <ListItem>
          <ListItemText primary="Text only" />
        </ListItem>
      </List>)

      expect(component).toMatchSnapshot('List with ListItemText snapshot')
    })
  })
})
