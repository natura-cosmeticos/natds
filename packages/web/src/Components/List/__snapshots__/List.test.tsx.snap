// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List components rendering variants should match to snapshot - empty list: List empty snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
/>
`;

exports[`List components rendering variants should match to snapshot - with ListItem button: List with ListItem button snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    Test
  </div>
</ul>
`;

exports[`List components rendering variants should match to snapshot - with ListItem: List with ListItem snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    Test
  </li>
</ul>
`;

exports[`List components rendering variants should match to snapshot - with ListItemAvatar: List with ListItemAvatar snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemAvatar-root"
    >
      <div
        className="MuiAvatar-root NatDsAvatar-root-1 MuiAvatar-circular MuiAvatar-colorDefault"
      >
        <i
          aria-hidden={true}
          className="material-icons MuiIcon-root natds-icons natds-icons-outlined-navigation-close NatDSIcon-root-3 NatDSIcon-root-4"
        />
      </div>
    </div>
  </li>
</ul>
`;

exports[`List components rendering variants should match to snapshot - with ListItemIcon: List with ListItemIcon snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <i
        aria-hidden={true}
        className="material-icons MuiIcon-root natds-icons natds-icons-outlined-navigation-close NatDSIcon-root-3 NatDSIcon-root-5"
      />
    </div>
  </li>
</ul>
`;

exports[`List components rendering variants should match to snapshot - with ListItemSecondaryAction: List with ListItemSecondaryAction snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemSecondaryAction-root"
    >
      <button
        aria-label="shopping"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <i
            aria-hidden={true}
            className="material-icons MuiIcon-root natds-icons natds-icons-outlined-navigation-close NatDSIcon-root-3 NatDSIcon-root-6"
          />
        </span>
      </button>
    </div>
  </li>
</ul>
`;

exports[`List components rendering variants should match to snapshot - with ListItemText: List with ListItemText snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <li
    className="MuiListItem-root MuiListItem-gutters"
    disabled={false}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        Text only
      </span>
    </div>
  </li>
</ul>
`;

exports[`List components rendering variants should match to snapshot - with ListSubheader: List with ListSubheader snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding MuiList-subheader"
>
  <li
    className="MuiListSubheader-root MuiListSubheader-sticky MuiListSubheader-gutters"
    id="subheader"
  >
    Subheader
  </li>
</ul>
`;
